packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -465,6 +546,75 @@ class NotifyingLogger extends Logger {
   }
 }
 
+/// A running application, started by this daemon.
+class AppInstance {
```

Comment:
This looks like a good place to keep connectivity info also, such as observatory/diagnostic ports and log streams.